src/Database/Type/DateTimeType.php

```diff
@@ -119,9 +127,31 @@ public function toDatabase($value, Driver $driver)
 
         $format = (array)$this->_format;
 
+        if ($this->dbTimezone !== null
+            && $this->dbTimezone !== $value->getTimezone()->getName()) {
+            $value->setTimezone(new DateTimeZone($this->dbTimezone));
```

Comment:
I think this needs to be `$value = $value->setTimezone(...)` to work with immutable datetime objects.

Reply:
Beware of `$value` in php 7.2 :D if that's still a thing

Reply:
It has only been a problem in loops.

Reply:
I'll fix this today.